src/envoy/http/authn/origin_authenticator.cc

```diff
@@ -56,10 +56,22 @@ void OriginAuthenticator::run() {
     }
     return;
   }
-  runMethod(credential_rule_.origins(0),
-            [this](const Payload* payload, bool success) {
-              onMethodDone(payload, success);
-            });
+  // Note to explain the change (the note will be removed after the code review)
```

Comment:
It's probably your JwtAuthenticatorCallbacks  keep a ref to the callback method. You can keep it by value instead, then this change is not necessary.

Reply:
Removed the notes and added an explanation that the unique_ptr callbackForRunMethod is used to deallocate the the MethodDoneCallback object when out of scope.